System/Console/ANSI/Windows/Foreign.hs

```diff
@@ -243,48 +246,61 @@ foreign import stdcall unsafe "windows.h FillConsoleOutputAttribute" cFillConsol
 foreign import stdcall unsafe "windows.h FillConsoleOutputCharacterW" cFillConsoleOutputCharacter :: HANDLE -> TCHAR -> DWORD -> UNPACKED_COORD -> Ptr DWORD -> IO BOOL
 foreign import stdcall unsafe "windows.h ScrollConsoleScreenBufferW" cScrollConsoleScreenBuffer :: HANDLE -> Ptr SMALL_RECT -> Ptr SMALL_RECT -> UNPACKED_COORD -> Ptr CHAR_INFO -> IO BOOL
 
+data ConsoleException = ConsoleException !ErrCode deriving (Show, Eq, Typeable)
+
+instance Exception ConsoleException
+
+-- Maybe someone can think of a better name for this function
```

Comment:
I think failIfFalse was pretty informative, while withErrorCode isn't. If you want to avoid a name clash, perhaps something like throwIfFalse, throwOnError, throwOnFailure?